Source/WTF/wtf/Vector.h

```diff
@@ -1781,7 +1781,7 @@ struct Mapper {
     using SourceItemType = typename CollectionInspector<SourceType>::SourceItemType;
     using DestinationItemType = typename std::invoke_result<MapFunction, SourceItemType&>::type;
 
-    static Vector<DestinationItemType> map(SourceType source, const MapFunction& mapFunction)
+    static Vector<DestinationItemType> map(const SourceType& source, const MapFunction& mapFunction)
```

Comment:
Will this cause us to lose the ability to efficiently move during the map? I am thinking of a test case like:

```c++
Vector<MoveOnlyType> source { a, b, c };
auto mapped = WTF::map(WTFMove(source), [] (MoveOnlyType&& item) {
    return TypeThatWrapsMoveOnlyType(WTFMove(item);
});
```

(I will freely admit I don't know if that worked previously).

Reply:
I don’t think it will; I think it will remove unneeded copying with no bad side effects. There’s a separate specialization below with for when the source is an rvalue reference. But I also think it’s not smart for me to modify this without adding tests, so I’ll take these unnecessary changes out of this pull request.

Reply:
Yes, I confirmed there are already move-only tests and this change does not break them. But I broke this change out into a separate pull request, where I have tests confirming it eliminates unnecessary copies.